.github/actions/build-gk/action.yaml

```diff
@@ -0,0 +1,86 @@
+name: 'build gk'
+description: 'Build GenomeKit conda package and save it under ~/conda-bld'
+inputs:
+  platform:
+    description: 'platform name used in the mamba installer url'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: get meta.yaml from conda-forge
+      shell: bash -l -e {0}
+      run: |
+        set -x
+        mkdir -p conda-recipe
+        curl -s -L -o conda-recipe/meta.yaml https://raw.githubusercontent.com/conda-forge/genomekit-feedstock/main/recipe/meta.yaml
+        set +x
+
+    - name: try to restore the mamba env
+      id: restore_mamba_cache
+      uses: actions/cache/restore@v4
+      with:
+        key: mamba-env-${{ runner.os }}-${{ runner.arch }}-${{ hashFiles('.github/workflows/run-tests.yaml', '.github/actions/**', 'conda-recipe/meta.yaml') }}
+        path: |
+          ~/micromamba
+          ~/.condarc
+          ~/bin/micromamba
+          ~/conda-bld
+
+    - name: install mamba
+      uses: ./.github/actions/install-mamba
+      with:
+        installer-url: https://micro.mamba.pm/api/micromamba/${{ inputs.platform }}/1.5.7
+
+    - name: create build env
+      shell: bash -l -e {0}
+      run: |
+        set -x
+        if [ ! -d "${HOME}/micromamba/envs/build" ]; then
+          micromamba create -yqn build -c conda-forge \
+            boa==0.16.0 mamba==1.5.7 conda==24.1.2 \
+            ccache
+        fi
+        set +x
+
+    - if: ${{ steps.restore_mamba_cache.outputs.cache-hit != 'true' }}
+      uses: ./.github/actions/configure-conda
+      with:
+        env: build
+
+    - name: save mamba gk ccache
+      id: save_mamba_gk_ccache
+      uses: actions/cache@v4
+      with:
+        key: gk-mamba-ccache-${{ runner.os }}-${{ runner.arch }}-${{ github.head_ref || github.ref_name }}-${{ github.sha }}
```

Comment:
is there a fallback? because this will just regenerate the cache everytime

Reply:
Good point. btw I noticed in the logs that the command doesn't actually create a `~/.ccache` dir, so nothing is cached.

Reply:
done and done

Reply:
ccache itself hashes the files.  the point is to only recompile files that have changed, not recompile everything.

in circleci, this is achieved by checking the git ref, then falling back branch, then arch.  not sure if github allows fuzzy matching like this

Reply:
Discussed offline and updated accordingly.